### ISSUE TITLE
[4.0.0-beta.17] Wrong signature for subscribeWithPriority in BackButtonEmitter 

### ISSUE THREAD

**Issue by TomDemulierChevret** [ionic-team/ionic#16535]
# Bug Report

**Ionic Info**
Run `ionic info` from a terminal/cmd prompt and paste the output below.

```
ionic info

Ionic:

   ionic (Ionic CLI)             : 4.5.0
   Ionic Framework               : @ionic/angular 4.0.0-beta.17
   @angular-devkit/build-angular : 0.10.7
   @angular-devkit/schematics    : 7.0.7
   @angular/cli                  : 7.0.7
   @ionic/angular-toolkit        : 1.2.0

Cordova:

   cordova (Cordova CLI) : 8.1.2 (cordova-lib@8.1.1)
   Cordova Platforms     : not available
   Cordova Plugins       : not available

System:

   NodeJS : v8.10.0 (C:\Program Files\nodejs\node.exe)
   npm    : 6.4.1
   OS     : Windows 10
```

**Describe the Bug**
The ``subscribeWithPriority`` method from the interface ``BackButtonEmitter`` (used for Android hardware back button) has the following signature :
```typescript
subscribeWithPriority(priority: number, callback: () => Promise<any> | void): void;
```
It's wrong because it actually returns a ``Subscription``.

**Expected Behavior**
This ``Subscription`` is really usefull if you want to make a temporary callBack to the hardware back button, therefore it should be available.

**Additional Context**
It was working in ``4.0.0-beta-16`` and the issue orginates from #16417.

Temporary workaround for those needing it :
```typescript
const backButtonSubscription: unknown = this.platform.backButton.subscribeWithPriority(9999, // your callback //);
<Subscription>backButtonSubscription).unsubscribe());
```

**Comment by olivermuc** [comment]
Just referencing this issue as it's also connected to the hardware back button handling.
One stone, two...
https://github.com/ionic-team/ionic/issues/16505

**Comment by TomDemulierChevret** [comment]
Even thought it's not in the changelog, I confirm that this is fixed by ``4.0.0-beta.18``.

**Comment by sasos90** [comment]
I still get this while running unit tests. Any feedback on this?
